### DISCUSSION TITLE
Noob questions for Maldives Hydrojet Pro

### DISCUSSION THREAD

**Discussion by SigmaPic** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#313] [Q&A]
Hello,

I am interested by this project for my spa.
I have the Maldives Hydrojet Pro model S200102 (60033).

I have a few questions.

What are the limitations of this project especially for my model if any.
Will I loose some features ?
Is there some known bugs that affect my model ?

You suggest two PCB design on the main page ?
What are the differences?
Which one is recommended?

Several times, I saw discussions about serial resistors (about 500ohms). Why is there so much discussions about these resistors?

Do I need to customize the pin config for my spa or it works as is.

I read a bit the code. I looked to the temperature slider that send a set target command. I follow the code up to the function that execute the set target command. Within this function , an up or down push is simulated. I wonder how does it work when the temperature you set with slider is quite different that the current one. It should simulate several push. isn't it ?

Thanks

**Comment by visualapproach** [comment]
> Hello,

Hello

> What are the limitations of this project especially for my model if any.

You won't loose weight running outside checking the spa temperature. Other than that, I can't think of anything.

> Will I loose some features ?
> Is there some known bugs that affect my model ?

Audio may be poor on 2021+ models. Don't know about your model.

> You suggest two PCB design on the main page ?
> What are the differences?
> Which one is recommended?

V2 is better. Ground plane and no crossing signal traces. Also more flexible.

> Several times, I saw discussions about serial resistors (about 500ohms). Why is there so much discussions about these resistors?

Because some 2021+ models needs them to get the display to not flash.


> Do I need to customize the pin config for my spa or it works as is.

You need to specify pcb and model in the file model.h. 6 wire I think has the same pin layout except for no54149e. That said, keep in mind that it's a DIY project with the possibility of not working due to bad soldering, components or non compatible pump. There has been success reported on S200102 though.

> I read a bit the code. I looked to the temperature slider that send a set target command. I follow the code up to the function that execute the set target command. Within this function , an up or down push is simulated. I wonder how does it work when the temperature you set with slider is quite different that the current one. It should simulate several push. isn't it ?

That's right

**Reply by SigmaPic** [reply]
Thank you for your prompt answer.

> Audio may be poor on 2021+ models. Don't know about your model.

Do you know why ?
Also, why do you need to connect the audio to the ESP. It's not possible to let it connected to the pump ?


> Because some 2021+ models needs them to get the display to not flash.

Any idea on how to check if they are needed ?
Other than check if display flash. I mean by monitoring the signal or something similar?
Are they needed on both side ?
Do you know why the display flash? 
Is there something input/output conflict?

> You need to specify pcb and model in the file model.h. 6 wire I think has the same pin layout except for no54149e.

How to check that pinout is the same ? Wire colors?


About the set target. I didn't see in the code where you manage the multi push to reach the target. Where is it managed ?

**Reply by visualapproach** [reply]
> Do you know why ?
> Also, why do you need to connect the audio to the ESP. It's not possible to let it connected to the pump ?

Different circuitry. Yes you can use the (not so) original sound.

> Any idea on how to check if they are needed ?
> Are they needed on both side ?
> Do you know why the display flash?

No. They are most likely only needed on the display port

> How to check that pinout is the same ? Wire colors?

Wire colors are unreliable. You'd need a logic analyzer to determine but as long as 5V and GND is right you won't damage anything. Check with a multimeter.

> About the set target. I didn't see in the code where you manage the multi push to reach the target. Where is it managed ?

It's a non blocking loop so it may not be easy to follow. But it's in the handleCommandQ() IIRC

**Comment by SigmaPic** [comment]
> Different circuitry. Yes you can use the (not so) original sound.

It' an output od the pump and an input of the display. Right ?
To keep original sound, I just need to bypass the esp board. Right ?
Each time the pump will get a simulated button push, a sound will be played, rigth ? 

> No. They are most likely only needed on the display port

About the resistors. If I'm rigth they are present on the pump PCB.
But there are some capacitors as well.
So I think it's recommended to mount both resistors and capacitors.
Right ?

> It's a non blocking loop so it may not be easy to follow. But it's in the handleCommandQ() IIRC

Ok. I din't catched how the set target command is rexecuted several time.
Need to analyse the code.

**Reply by visualapproach** [reply]
Yes just connect the audio pins from pump to display. The CIO can't tell if you press a physical button or if you do it from the web ui so it will beep unless you disable certain buttons in the code. See build instructions.
Using both resistors and capacitors is the proper way to make a low pass filter to suppress spikes in the signal but the wires themself acts as small caps so it's not always necessary to put them there. It's been trial and error and not super scientific. Although I calculated the RC cutoff frequency to be somewhere in the right ballpark.

**Reply by SigmaPic** [reply]
> Yes just connect the audio pins from pump to display. The CIO can't tell if you press a physical button or if you do it from the web ui so it will beep unless you disable certain buttons in the code. See build instructions.

Maybe I'll try to modify the PCB routing to allow a bypass for the audio signal.

> Using both resistors and capacitors is the proper way to make a low pass filter to suppress spikes in the signal but the wires themself acts as small caps so it's not always necessary to put them there. It's been trial and error and not super scientific. Although I calculated the RC cutoff frequency to be somewhere in the right ballpark.

If I'm rigth, there is also such low pass filter on the CIO side. We can see a capa and a resistor near the connector.
You said that frequency is 22kHz => I will check the cutoff frequency.

In the excel file you said:

> Active LOW, shift out on falling edge. Latch on rising edge. CIO seems bad at this, or maybe there is noise on the line. This can be filtered out in SW. Clock speed on the CIO is about 22KHz

What do you means by : CIO seems bad at this, or maybe there is noise on the line. This can be filtered out in SW.
Can you explain ?

**Reply by visualapproach** [reply]
Meaning the cio switched state on wrong edge occasionally
